<launch>
  <!-- Arguments needed to set up team -->
  <!-- team: YELLOW_TEAM or BLUE_TEAM -->
  <arg name="team"/>
  <!-- side: LEFT_SIDE or RIGHT_SIDE -->
  <arg name="side"/>
  <!-- Topic name where the ball position is published -->
  <arg name="ball_topic"/>

  <!-- Set parameters that some nodes may need to read on initialization  -->
  <param name="team" type="string" value="$(arg team)" />
  <param name="side" type="string" value="$(arg side)" />

  <!-- Nodes that compose the system -->
  <node pkg="futrobotros" type="localization_node" name="localization">
    <remap from="ball_position" to="$(arg ball_topic)"/>
  </node>
  <node pkg="futrobotros" type="strategy_node" name="strategy">
    <remap from="ball_position" to="$(arg ball_topic)"/>
  </node>
  <node pkg="futrobotros" type="obstacle_avoidance_node" name="obstacle_avoidance">
    <remap from="ball_position" to="$(arg ball_topic)"/>
  </node>
  <node pkg="futrobotros" type="control_node" name="control">
    <remap from="team_references" to="team_corrected_references"/>    
  </node>

  <!-- Tf broadcasters to visualize poses in tf -->
  <node pkg="futrobotros" type="team_pose_to_tf" name="team_tf_broadcaster">
    <param name="prefix" type="string" value="robot" />
    <remap from="team_poses_input" to="team_poses"/>
  </node>
  <node pkg="futrobotros" type="team_pose_to_tf" name="ref_tf_broadcaster">
    <param name="prefix" type="string" value="ref" />
    <remap from="team_poses_input" to="team_references"/>
  </node>
  <node pkg="futrobotros" type="team_pose_to_tf" name="corr_ref_tf_broadcaster">
    <param name="prefix" type="string" value="corr_ref" />
    <remap from="team_poses_input" to="team_corrected_references"/>
  </node>

</launch>
